common/kvstore/src/main/java/org/apache/spark/util/kvstore/InMemoryStore.java

```diff
@@ -194,21 +228,44 @@ public int size() {
     }
 
     @SuppressWarnings("unchecked")
-    public <T> InMemoryView<T> view(Class<T> type) {
-      Preconditions.checkArgument(ti.type().equals(type), "Unexpected type: %s", type);
-      Collection<T> all = (Collection<T>) data.values();
-      return new InMemoryView<>(type, all, ti);
+    public InMemoryView<T> view() {
+      return new InMemoryView<>(data.values(), ti);
     }
 
+    @SuppressWarnings("unchecked")
+    private static <T> Predicate<? super T> getPredicate(
+        KVTypeInfo.Accessor getter,
+        Collection keys) {
+      if (Comparable.class.isAssignableFrom(getter.getType())) {
+        HashSet set = new HashSet(keys);
+
+        return (value) -> set.contains(keyFromValue(getter, value));
+      } else {
+        HashSet<Comparable> set = new HashSet<>(keys.size());
+        for (Object key : keys) {
+          set.add(asKey(key));
+        }
+        return (value) -> set.contains(asKey(keyFromValue(getter, value)));
+      }
+    }
+
+    private static Object keyFromValue(KVTypeInfo.Accessor getter, Object value) {
+      try {
+        return getter.get(value);
+      } catch (Exception e) {
+        throw Throwables.propagate(e);
```

Comment:
cf. prior deprecation comment

Reply:
Yeah -- I copied this code from elsewhere in InMemoryStore because getter.get is, afaict, everywhere used where exceptions can't be rethrown.  Or something similarly as stupid :|  I was not amused.
get() is defined as throwing Exception, so I can't use one of the simpler alternatives, and if I touch this code, I'm going to stop allowing exceptions being thrown out, and force the implementation to do this propagation where necessary.